kratos/utilities/search_utilities.h

```diff
@@ -73,22 +74,40 @@ class SearchUtilities
 
     /**
      * @brief Check if a point is inside a bounding box
+     * @details Bounding box class implementation
+     * @param rBoundingBox The bounding box
+     * @param rCoords The point
+     * @return true if the point is inside the bounding box
+     */
+    static bool PointIsInsideBoundingBox(
+        const BoundingBox<Point>& rBoundingBox,
+        const array_1d<double, 3>& rCoords
+        );
+
+    /**
+     * @brief Check if a point is inside a bounding box
+     * @details Bounding box array of 6 doubles implementation
      * @param rBoundingBox The bounding box
      * @param rCoords The point
      * @return true if the point is inside the bounding box
      */
     static bool PointIsInsideBoundingBox(
         const BoundingBoxType& rBoundingBox,
         const array_1d<double, 3>& rCoords
-        )
-    {
-        // The Bounding Box should have some tolerance already!
-        if (rCoords[0] < rBoundingBox[0] && rCoords[0] > rBoundingBox[1])           // check x-direction
-            if (rCoords[1] < rBoundingBox[2] && rCoords[1] > rBoundingBox[3])       // check y-direction
-                if (rCoords[2] < rBoundingBox[4] && rCoords[2] > rBoundingBox[5])   // check z-direction
-                    return true;
-        return false;
-    }
+        );
+
+    /**
+     * @brief This method checks if a point is inside any bounding box of the global bounding boxes considering a certain tolerance
+     * @param rBoundingBox The bounding box
+     * @param rCoords The coordinates of the point
+     * @param Tolerance The tolerance
+     * @return True if the point is inside the bounding box
+     */
+    static bool PointIsInsideBoundingBoxWithTolerance(
```

Comment:
the function's signature is different and it's pretty straightforward that the extra argument is a tolerance, so this function could just be an overload of `PointIsInsideBoundingBox`. No need to explain the argument in the function's name too.

Reply:
huuuuum, true. I will think about changing it

Reply:
Done